### ISSUE TITLE
[device-port] [halium-9.0] [dreamlte] Samsung Galaxy S8

### ISSUE THREAD

**Issue by Jo-Blade** [Halium/projectmanagement#292]
Tree: 

- [x] Create manifest [samsung_dreamlte.xml](https://github.com/Halium/halium-devices/pull/331/files#diff-3e591ad619748e66e78bd398710d6e54e1ecb691e239c003d6f23147ef5be471) 
- [x] Boot image and system image build successfully
- [x] Device boots into rootfs, `usb: Manufacturer: GNU/Linux Device` appears in `dmesg` on host.
- [x] LXC container starts and does not crash
- [ ] libhybris tests
  * [ ] test_gps
  * [ ] test_hwcomposer
  * [ ] test_lights
  * [ ] test_vibrator
  * [ ] test_wifi
  * [ ] test_sensors
  * [ ] test_audio
  * [ ] test_camera
  * [ ] test_input
  * [ ] test_nfc
  * [ ] test_recorder

IMPORTANT NOTE : usb is working only after adding the usb-tethering file to /usr/bin that can be done in twrp recovery after flashing the system.img by mounting the /data/rootfs then copying the usb-tethering binary to /usr/bin and chmod 755
In all cases you should be able to bring up lomiri without it (but you won't be able to connect ssh through usb, wifi only)
nota bene: do not forgot to give a link to the usb-tethering file here…

**Comment by Jo-Blade** [comment]
First status update, firstly thanks to @abkro to help me to create the manifest file (and doing the most of the work, you should take a sight to his repository).

I've successfully managed to build and boot the system image on my samsung S8. Lomiri started and opengl apps are working (I've tried blender for example). I've successfully connected the phone to my TV using the usb-c port and also a mouse and a keyboard. Furthermore, the wifi is working but you may have to check the repos on github to take the correct udev rules.

However, there is a lot of bugs and random crash of the UI (it should restart automatically after a few minutes) who make it no very usable for now, and the sound is not working at all. (the bluetooth doesn't work too)

For more information, you can join the ubport porting telegram group or contact me through github or my personal matrix account @pisento1:pisento.tk

**Comment by Jo-Blade** [comment]
![IMG_20220328_124521](https://user-images.githubusercontent.com/59778661/162533252-3b0a10b4-7c0a-4481-aad0-1933bda60bf0.jpg)
![lomiri on tv](https://user-images.githubusercontent.com/59778661/162533521-0e85d576-3651-4845-bf81-9414b9254e4c.jpg)


here some pictures to give you more hype

**Comment by mgood7123** [comment]
nice

**Comment by abkro** [comment]
I picked up my work on the S8 after a 2 month break. I see there have been made attempts at Halium10, but I will continue my work on Halium9 for now.

One of the weird things with this device has been how the udev rules file generated on the device after first build allowed the UI to start, but the device would also hang for minutes without responding. I got past this by using a udev rules file generated on the host based on the file ueventd.samsungexynos8895.rc in the source tree. This resulted in a very much shorter udev rules file, but the device seemingly worked normally.

I have now determined that the problem is connected to one or more (probably two) of the lines associated with /dev/*binder. If I use the original file generated on the device, but omit these three lines, the device apparently reponds normally.

So far, I have not commited these changes to my repo because I am still trying to determine exactly what service is missing and causing the hanging behavior when I use the complete udev rules.

**Comment by abkro** [comment]
I should also mention that although the various tests mentioned at the top of the page don't work as they should, I can still confirm that sensors work, including:
- accelerometer
- compass
- gyroskope
- rotation
- orientation
- ambient light
- proximity

Also, gps works.

Bluetooth is active, but something prevents the device from connecting properly to any bluetooth device. It connects, then immediately loses the connection, by the looks of it.

Wifi works and with the shorter udev rules file mentioned above, it also reconnects to a previously used SSID after a reboot, but with the longer quasi-complete udev rules file, it does not reconnect automatically.

Today, I found out how to enable the automatic screen brightness toggle button, but it still does not adjust the brightness automatically when enabled, so I am not going to commit this change yet.

**Comment by Jo-Blade** [comment]
Yes I've tried to port on halium 10 first, but halium 10 on non treble device is truly a mess, never successfully started the lxc container so I left this for the moment. I will update the working sensor this evening according to your post.

I just wanted to add that even if your udev configuration is really better, I had to go back on the default autogenerated one to make pulseaudio able to start, so maybe these lines were related to audio system…
I currently just succeed to start pulseaudio but with no working output (I've modified some files, I will post them here if I get good news in the future), the error message is "E: [droid-sink-fast] droid-sink.c: failed to write stream (-22)", I founded no relevant information on the web for now.

**Comment by abkro** [comment]
What do you mean by "default autogenerated" udev rules? I'm not aware of anything that autogenerates...

**Comment by manicpixiedreamboy** [comment]
Hey! Managed to build Halium 9 using your repos

Regarding the udev rules, is there any solution? It still hangs after a minute or two...

**Comment by ChampVibing** [comment]
can we install this now or de I have to wait a bit longer?

**Comment by Jo-Blade** [comment]
You can try it now if you want, I don't work on it currently. But, you may be disappointed because there is a lot of issue, a lot of crash and there is no sound… (I can't be actually usable for now)

**Comment by manicpixiedreamboy** [comment]
@mcgodnl the port is working poorly - no sound and rebooting after some time (e.g. like every 5-10 minutes). also i suppose hardware acceleration is not working yet - therefore many native apps either don't work at all or cause extra reboots
I am thinking about starting from scratch and trying to build sailfish os, but I don't have much time now (

**Comment by Kioro404** [comment]
Hello @Jo-Blade, I would like to port Ubuntu Touch to my Samsung S8 (DreamLTE) device with an Exynos 8895 processor, with the most recent version of LineageOS and if possible with Halium 11.0. However, I would like to have the latest corrections to the udev rules that you mentioned and that you did not make public so I can continue with your hard work.

**Comment by Jo-Blade** [comment]
Hello @Kioro404 , I'm happy to see there is still people interrested with this project, but unfortunately I have nothing new.

I haven't worked anymore on this project due to a lack of time with my studies. I still have my device however but I reverted it to lineageos for a long while.

I am sorry to not be able to help you more
